src/qt/bitcoin.cpp

```diff
@@ -431,6 +433,32 @@ int GuiMain(int argc, char* argv[])
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
 #endif
 
+#if defined(WIN32)
+#else
+    node->setupServerArgs();
+    SetupUIArgs();
+    std::string error;
+    if (!gArgs.ParseParameters(argc, argv, error))
```

Comment:
Instead of duplicating code, perhaps move it to a new function.

Reply:
Agree. I really dislike duplicating all this code. I'm also not convinced that moving `ParseParameters` to another place in the initialization sequence won't interfere with some other part of the initialization sequence. It's quite a fragile sequence, carefully balanced to work for all use cases.